src/Functions/CMakeLists.txt

```diff
@@ -9,10 +9,11 @@ add_headers_and_sources(clickhouse_functions .)
 list(REMOVE_ITEM clickhouse_functions_sources IFunction.cpp FunctionFactory.cpp FunctionHelpers.cpp)
 list(REMOVE_ITEM clickhouse_functions_headers IFunction.h FunctionFactory.h FunctionHelpers.h)
 
-add_library(clickhouse_functions ${clickhouse_functions_sources})
+add_library(clickhouse_functions_obj OBJECT ${clickhouse_functions_sources})
```

Comment:
Just for my understanding: Is it necessary for this approach to work to change the static/dynamic library "clickhouse_functions_obj" into an object library?

Reply:
It's the easiest way. Otherwise we need to carefully examine the linking process to make sure `register*` symbols are picked up by the linker.